plc4j/protocols/driver-bases/base/src/main/java/org/apache/plc4x/java/base/connection/AbstractPlcConnection.java

```diff
@@ -57,6 +67,42 @@ public boolean canSubscribe() {
         return false;
     }
 
+    /**
+     * The default implementation uses the {@link #ping(int)} method.
+     * Drivers that want to implement a more specific version have to overide it.
+     */
+    @Override
+    public boolean isConnected() {
+        return ping(PING_TIMEOUT_MS);
```

Comment:
could we prepend a `channel.isActive()` here as well?

Reply:
with a or so we don't try to send a ping in first place if we don't have an active channel

Reply:
will add a todo, as I'm unsure if we should do that or not...